source_models.go

```diff
@@ -0,0 +1,119 @@
+package main
```

Comment:
Code in this file should be moved to the other files (packages in the future) to be flexible and aligned with golang philosophy.
In short, golang interfaces are not intended to be used like interfaces in OOP languages, they are more like duck-typing in dynamic languages. There couple rules of thumb here:
  - interfaces are declared in the place of usage (client code), often such interfaces are private for client package;
  - functions should accept interfaces and return structs (it is very good rule for New functions espessially).

This is an article i've googled which can give the idea: https://alok87.in/blog/interfaces-go

Therefore:
1) All interfaces should be defined in the place of usage: 
  - `Source` should move in app.go 
  - `SourceUser`, `SourceGroup` in diff.go
  - `SourceGroupWithMembers` also in diff.go
2) AzureUser, AzureGroup and everything Azure should be in azure*.go
3) NewSource{User,Group} functions shouldn't exists. It should be NewAzureUser (and NewLdapUser in the future) in azure*.go which should return AzureUser _struct_ (not SourceUser interface).